fe/src/main/java/org/apache/doris/qe/ShowExecutor.java

```diff
@@ -302,44 +304,77 @@ private void handleShowEngines() {
         resultSet = new ShowResultSet(showStmt.getMetaData(), rowSet);
     }
 
-    // Handle show function
-    private void handleShowFunction() throws AnalysisException {
-        ShowFunctionStmt showStmt = (ShowFunctionStmt) stmt;
-
+    // Handle show functions
+    private void handleShowFunctions() throws AnalysisException {
+        ShowFunctionsStmt showStmt = (ShowFunctionsStmt) stmt;
         Database db = ctx.getCatalog().getDb(showStmt.getDbName());
         if (db == null) {
             ErrorReport.reportAnalysisException(ErrorCode.ERR_BAD_DB_ERROR, showStmt.getDbName());
         }
-        List<Function> functions = db.getFunctions();
+        List<Function> functions = showStmt.getIsBuiltin() ? ctx.getCatalog().getBuiltinFunctions() :
+            db.getFunctions();
 
-        List<List<String>> rowSet = Lists.newArrayList();
+        List<List<Comparable>> rowSet = Lists.newArrayList();
         for (Function function : functions) {
-            List<String> row = Lists.newArrayList();
-            // signature
-            row.add(function.getSignature());
-            // return type
-            row.add(function.getReturnType().getPrimitiveType().toString());
-            // function type
-            // intermediate type
-            if (function instanceof ScalarFunction) {
-                row.add("Scalar");
-                row.add("NULL");
-            } else {
-                row.add("Aggregate");
-                AggregateFunction aggFunc = (AggregateFunction) function;
-                Type intermediateType = aggFunc.getIntermediateType();
-                if (intermediateType != null) {
-                    row.add(intermediateType.getPrimitiveType().toString());
-                } else {
+            List<Comparable> row = Lists.newArrayList();
```

Comment:
I think the `row` built here can be implemented in `Function`. eg:

function.getInfo(row, isVerbose);